wire-ios-sync-engine/Source/UserSession/ZMUserSession.swift

```diff
@@ -711,17 +711,20 @@ extension ZMUserSession: ZMSyncStateDelegate {
 
     public func didFinishSlowSync() {
         managedObjectContext.performGroupedBlock { [weak self] in
-            self?.hasCompletedInitialSync = true
-            self?.notificationDispatcher.isEnabled = true
+            guard let self else { return }
 
-            if let context = self?.managedObjectContext {
-                ZMUserSession.notifyInitialSyncCompleted(context: context)
-            }
+            self.hasCompletedInitialSync = true
+            self.notificationDispatcher.isEnabled = true
+            ZMUserSession.notifyInitialSyncCompleted(context: managedObjectContext)
         }
 
-        let selfClient = ZMUser.selfUser(in: syncContext).selfClient()
-        if selfClient?.hasRegisteredMLSClient == true {
-            Task {
+        Task {
+            let hasRegisteredMLSClient = await syncContext.perform {
+                let selfClient = ZMUser.selfUser(in: self.syncContext).selfClient()
+                return selfClient?.hasRegisteredMLSClient == true
+            }
```

Comment:
Because we now call `didFinishSlowSync()` also from another thread through async await und thus the access of `ZMUser.selfUser(in: syncContext).selfClient()` crashed.

Reply:
you should call it from syncContext

Reply:
Please review my changes about the queues